File: maplejh/Programmers/실패율.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42889
def solution(N, stages):
    answer = []
    cur = [0] * (N + 2)  # 현재 진행중인 사람
    for s in stages:
        cur[s] += 1
    ppl = cur[N + 1]
    for i in range(N, 0, -1):
        ppl += cur[i]
        if ppl:
            answer.append((cur[i] / ppl, i))
        else:
            answer.append((0, i))
    answer = [c[1] for c in sorted(answer, key=lambda x: (-x[0], x[1]))]
    return answer

File: maplejh/Programmers/실패율.java
//https://school.programmers.co.kr/learn/courses/30/lessons/42889


class Solution {
    public int[] solution(int N, int[] stages) {
        int[] answer = new int[N];
        int player = stages.length;
        int[] cur = new int[N + 1];
        double[] failure = new double[N];
        for (int s : stages) {
            cur[s - 1] += 1;
        }
        int cnt = player;
        for (int i = 0; i < N; i++) {
            failure[i] = (double) cur[i] / cnt;
            cnt -= cur[i];
            answer[i] = i + 1;
        }
        int a, b, tmpi;
        double tmpd;
        for (int i = 0; i < N; i++) {
            for (int j = i + 1; j < N; j++) {
                if (failure[i] < failure[j] || (failure[i] == failure[j] && answer[i] > answer[j])) {
                    tmpd = failure[i];
                    tmpi = answer[i];
                    failure[i] = failure[j];
                    answer[i] = answer[j];
                    failure[j] = tmpd;
                    answer[j] = tmpi;
                }
            }
        }
        return answer;
    }
}
